packages/common-all/src/types/typesv2.ts

```diff
@@ -169,10 +170,25 @@ export type DNodePropsDict = {
   [key: string]: DNodeProps;
 };
 
+/**
+ * Map of noteId -> noteProp
+ */
 export type NotePropsDict = {
   [key: string]: NoteProps;
 };
 
+/**
+ * Map of noteFname -> list of noteIds. Since fname is not unique across vaults, there can be multiple ids with the same fname
+ */
+export type NotePropsByFnameDict = {
+  [key: string]: string[];
+};
+
+export type NotePropsFullDict = {
```

Comment:
name is a little obtuse, especially since this doesn't return `NoteProps` by a dictionary instead. rename to `NoteDicts` instead?

Reply:
I'd also recommend a docstring that links to the utils class to remind anyone using this that they should be accessing through that class.

Reply:
Fixed